src/transformers/modeling_utils.py

```diff
@@ -309,7 +309,13 @@ def shard_checkpoint(
     current_block_size = 0
     total_size = 0
 
-    for key, weight in state_dict.items():
+    unique_storage_to_save = {
+        (weight.storage().data_ptr(), weight.storage().nbytes()): key for key, weight in state_dict.items()
```

Comment:
For safetensors saving, we use the tuple `(tensor.data_ptr(), tensor.device, tensor.shape, tensor.stride())` to identify the tensor. Is this better as unique identifier?
Would prefer to have one source of truth (and maybe we could have a util function that gets a unique identifier from a tensor)

Also does `.storage()` work from PyTorch 1.8 and onward?

Reply:
Ah good point! I think we need to add device as well. Basically the difference with safetensors is I need storage identifier while safetensors searches for a tensor identifier. The difference is for example `.transpose()` has the same storage, but it's clearly not the same tensor (shape differs)